Automattic/wp-calypso
# Strip subfolders from base URL in site preview URL field

#### Proposed Changes

Currently in the site preview URL field, the base URL is defined as `this.props.site.URL`, which can be problematic as this includes the subfolder of a WP installation for Jetpack connected sites

The `pathname` property is appended to this base URL, and in instances where a Jetpack site has WordPress installed in a subfolder, this creates a duplicate subfolder path, i.e., `https://foo.com/sub/sub/`

This PR uses the `URL` constructor and `origin` property to strip the subfolder from the base URL, preventing the `pathname` operator from creating a duplicate subfolder in the URL displayed in site preview URL field.

Big thanks to @fullofcaffeine for pointing me to to the URL.origin property: https://developer.mozilla.org/en-US/docs/Web/API/URL/origin

#### Testing Instructions



* Create a new WP installation in a subfolder and connect it to Jetpack. You can use https://jurassic.ninja/specialops/ with the "Launch Multisite on subdirs" option. Then add a site in the Network Admin Sites screen: https://wordpress.org/support/article/network-admin-sites-screen/
* Navigate to https://wordpress.com, click "Switch Site" in upper left, and select your newly created site
* Preview the site by clicking the Site URL/Title in the upper left hand corner of Calypso
* Verify that URL is properly updated with the correct path when clicking on pages/posts in the site's preview
* Test on Simple, Atomic, and Jetpack connected sites (with root WP install)

#### BEFORE:
<img width="1264" alt="before" src="https://user-images.githubusercontent.com/65082164/190023177-bc0c9f68-db05-495f-9eb7-4e9e9eda52e2.png">


#### AFTER:
<img width="1263" alt="after" src="https://user-images.githubusercontent.com/65082164/190023199-2c584807-5cfb-4d5b-bce8-08c3ab5112cf.png">


#### Pre-merge Checklist






Related to #67563
